src/client/pkg/tracing/extended/extended_trace.go

```diff
@@ -2,141 +2,137 @@ package extended
 
 import (
 	"context"
-	"encoding/base64"
-	"fmt"
 	"os"
+	"time"
 
 	"github.com/pachyderm/pachyderm/src/client/pkg/tracing"
 	col "github.com/pachyderm/pachyderm/src/server/pkg/collection"
 
 	etcd "github.com/coreos/etcd/clientv3"
 	opentracing "github.com/opentracing/opentracing-go"
-	"github.com/opentracing/opentracing-go/ext"
 	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
 	log "github.com/sirupsen/logrus"
 	"google.golang.org/grpc/metadata"
 )
 
 const (
-	// traceCtxKey is the grpc metadata key whose value is a ExtendedTrace
-	// identifying the current RPC/commit
-	traceCtxKey = "commit-trace"
+	// traceCtxKey is the grpc metadata key whose value is a serialized
+	// ExtendedTrace proto tied to the current CreatePipeline request. In a grpc
+	// quirk, this key must end in '-bin' so that the value (a serialized
+	// timestamp) is treated as arbitrary bytes and base-64 encoded before being
+	// transmitted (see
+	// https://github.com/grpc/grpc-go/blob/b2c5f4a808fd5de543c4e987cd85d356140ed681/Documentation/grpc-metadata.md)
+	traceCtxKey = "pipeline-trace-duration-bin"
```

Comment:
I'd be careful about calling this a "context key", because those are something different; see `context.WithValue` and friends.

Reply:
Done. I went with traceMDKey here—hopefully that's clearer. We can also easily change this later since it's private.